master/internal/user/postgres_long_lived_tokens.go

```diff
@@ -0,0 +1,137 @@
+package user
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/o1egl/paseto"
+	"github.com/pkg/errors"
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+const (
+	// TokenExpirationDuration is how long a newly created long lived token is valid.
+	TokenExpirationDuration = 30 * 24 * time.Hour
+)
+
+// LongLivedTokenOption is the return type for WithTokenExpiresAt helper function.
+// It takes a pointer to model.LongLivedToken and modifies it.
+// It’s used to apply optional settings to the LongLivedToken object.
+type LongLivedTokenOption func(f *model.LongLivedToken)
+
+// WithTokenExpiresAt function will add specified expiresAt (if any) to the long lived token table.
+func WithTokenExpiresAt(expiresAt *time.Time) LongLivedTokenOption {
+	return func(s *model.LongLivedToken) {
+		s.ExpiresAt = *expiresAt
+	}
+}
+
+// DeleteAndCreateLongLivedToken creates a row in the long lived token table.
+func DeleteAndCreateLongLivedToken(
```

Comment:
question (non-blocking): is there another way we could convey this concept more concisely? Would `Overwrite` or `Replace` be clear? If not and `DeleteAndCreate` is more explicitly stating exactly what is being performed here that's absolutely fair, just wanted to see if alternatives were considered here

Reply:
Which one would you vote for - `ReplaceLongLivedToken` or `ReissueLongLivedToken` ?
We can change it to include `*LongLivedAccessToken` once we have product consensus.